src/index.js

```diff
@@ -114,37 +115,17 @@ module.exports = Class.extend({
 
       this._pendingAssociations = _.chain(functions)
          .reduce(function(memo, fnDef, fnName) {
-            var distName, evtType, dist;
-
             if (!fnDef.lambdaAtEdge) {
                return memo;
             }
 
-            distName = fnDef.lambdaAtEdge.distribution;
-            evtType = fnDef.lambdaAtEdge.eventType;
-            dist = template.Resources[distName];
-
-            if (fnDef.lambdaAtEdge) {
-               if (!_.contains(VALID_EVENT_TYPES, evtType)) {
-                  throw new Error('"' + evtType + '" is not a valid event type, must be one of: ' + VALID_EVENT_TYPES.join(', '));
-               }
-
-               if (!dist) {
-                  throw new Error('Could not find resource with logical name "' + distName + '"');
-               }
-
-               if (dist.Type !== 'AWS::CloudFront::Distribution') {
-                  throw new Error('Resource with logical name "' + distName + '" is not type AWS::CloudFront::Distribution');
-               }
-
-               memo.push({
-                  fnLogicalName: self._provider.naming.getLambdaLogicalId(fnName),
-                  distLogicalName: fnDef.lambdaAtEdge.distribution,
-                  fnCurrentVersionOutputName: self._provider.naming.getLambdaVersionOutputLogicalId(fnName),
-                  eventType: evtType,
+            if (fnDef.lambdaAtEdge.constructor === Array) {
```

Comment:
Rather than directly checking against an object's constructor, prototype, or `typeof`, we try to use functions like [`_.isArray`](http://underscorejs.org/#isArray) when possible. It tends to be a little easier to read and help to prevent subtle type checking bugs.